### ISSUE TITLE
All form validation error messages should be vocalized as soon as they happen

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13676]
**Describe the bug**
- Consistently all throughout, none of the form validation error messages are currently being vocalized by any screen reader software.

**Tell us the version of Spartacus**
- Library version: 4.0.0.

**To Reproduce**
1. Open any Screen Reader (VO, NVDA, JAWS).
2. Go to the Sign-In page
3. Introduce invalid email in the form and leave the password field blank.
4. See errors, and hear the absence of vocalization of those same errors.

**Non-thorough list of UIs where this is needed:**
- Login
- Registration
- Coupon field
- Payment details
- My Account > Personal details
- My Account > Password
- My Account > Email Address

**Expected behavior**
- All form validation error messages get vocalized as soon as they happen.
- Since this need the user immediate attention, `aria-live="assertive" aria-atomic="true"`
- E.g.     `<cx-form-errors  aria-live="assertive" aria-atomic="true"></cx-form-errors>`

**Screenshots**

![Screen Shot 2021-09-09 at 8 31 02 PM](https://user-images.githubusercontent.com/15113219/132779864-f025b481-cac4-486d-b09b-61a8afbcfbed.png)

**Example of potential solution**

![Screen Shot 2021-09-09 at 7 59 38 PM](https://user-images.githubusercontent.com/15113219/132779887-f27442ed-14b9-430a-a1d7-fa91b21d6dde.png)

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

✅ Login
❌ Registration -> See comment below
✅ Coupon field
❌ Payment details -> See comment below
✅ My Account > Personal details
✅ My Account > Password
✅ My Account > Email Address

For me, it's missing the voice error messages for 
- the password fields on registration page. If you don't enter a good format, I don't hear it say it, it only displays it on the bottom. 
It will say it only if I move away from that field, and then focus on it again. But if I don't focus on it again, and I move to the next field, and I click on the "Register" button, then I don't know what's wrong because I don't hear anything. 

- Payment Details page during checkout. If you input an invalid expiry date (in the past), it does not voice the error message. 
![expiryDate](https://user-images.githubusercontent.com/45409515/146254887-14a5ff8f-36db-4f20-b0df-391f3e71036a.png)

@developpeurweb Can you confirm the above please.

**Comment by KateChuen** [comment]
After talking to Miguel and Saad, we agreed to reopen this ticket to tackle the password issue on the registration page. 

However, the issue about the payment details will be tackled in ticket [14371](https://github.com/SAP/spartacus/issues/14371)

**Comment by developpeurweb** [comment]
This works on VO, but unfortunately not for JAWS/NVDA due to tags not being preloaded as a required condition for aria-live to work. Since we still don't have a solution for preloading tags, **let's try the `role="alert"` approach here**, which we know is detected by JAWS/NVDA/VO. This might be a bit aggressive for some SR users but it's justified in the sense that **validation errors require immediate attention** from users to be able to continue their UI journey.

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using yarn
More extensive testing (on other screenreaders and devices) will be done with RC.

- Login screen  ✅
- Registration Page ✅ 
- Coupon field ✅ 
- Quick order field ✅ 
- Payment details ✅ 
- My Account -> Personal Details ✅ 
- My Account -> Email Address ✅ 
- My Account -> Password ✅

**Comment by jasdavistopcon** [comment]
Is there a way to overwrite the built in message 'this is not a valid email format'?
